### ISSUE TITLE
Not working for "polarplots"

### ISSUE THREAD

**Issue by MarvNum** [matlab2tikz/matlab2tikz#950]
I tried to use matlab2tikz for creating a tikz from a polarplot. Unfortunately the program has a problem with the polar axes class.
"Error using get
There is no DataAspectRatioMode property on the PolarAxes class."
Can anyone help me. Wasn't able to fix it myself ;-(

**Comment by egeerardyn** [comment]
Thank you for reporting this! 😄 

I don't know whether this will fix everything, since it is probably strongly related to #410 (i.e. we lack proper support for special plots), but you might want to try replacing
```matlab
get(handle, 'DataAspectRatioMode')
```
in the code with something like
```matlab
getOrDefault(handle, 'DataAspectRatioMode','')
```
That could at least get you past those errors 😄, but I'm not sure how far you will get.

**Comment by mawwinius** [comment]
I ran into the same issue and tried your suggestion. There are many places where you had to place this workaround (didn't go through all of them) and it will probably break a lot.

There are also things like axis limits that are broken, so the plot won't be fine in the end.

**Comment by MarvNum** [comment]
Hey everyone. For myself I solved the issue writting my own function "mypolarplot" that fixes the axes of a standardplot to look like the polarplot if anyone is interested I can share it ;)

**Comment by mawwinius** [comment]
yes, I'm interested :) Would be similar to the old "polar" command i guess?

**Comment by MarvNum** [comment]
function [] =  myPolarPlot(maxY, myRad, myAngle)

if nargin == 1
    myRad = 'r/R';
    myAngle = '\Psi';
end

%figure();
%create circle;
hold on;
[coefficient,exponent] = ...
strread(strrep(sprintf('%E',maxY),'E','#'),'%f#%f');
coeffRound = ceil(coefficient);
if coeffRound < 2
     nocircs = coeffRound *5;
   delta = coeffRound/(nocircs) * 10^exponent;  
elseif    coeffRound < 4
   nocircs = coeffRound *2;
   delta = coeffRound/(nocircs) * 10^exponent;
elseif coeffRound > 6 
    coeffRound = ceil(coeffRound/2)*2;
    nocircs = coeffRound/2;
    delta = coeffRound/(nocircs) * 10^exponent;
else
    nocircs = coeffRound;
    delta = 10^exponent;
end

phiText = 10*pi/180;

r = 0;

    

for i = 1:nocircs
    r = r + delta;

            phi = linspace(0,2*pi,1000);
            plot(r*sin(phi),r*cos(phi),':black');
    
    if abs(exponent) > 4
    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.0e',r));
    else
    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), num2str(r));
    end
%    if exponent == 0
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.0f',r));
%    elseif exponent == 1
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.0f',r));
%    elseif exponent == 2
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.0f',r));
%    elseif exponent == 3
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.0f',r));
%    elseif exponent == -1
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.1f',r));
%    elseif exponent == -2
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.2f',r));
%    else
%    text((r+0.2*delta)*(sin(phiText)+0.05), (r+0.2*delta)*cos(phiText), sprintf('%0.0e',r));
%    end
    hold on;
end
            plot(r*sin(phi),r*cos(phi),'black');

noangles = 12;
phi = 0;

while phi < 2*pi
    phi = phi + 2*pi/noangles;
    plot([0 r*sin(phi)],[0 r*cos(phi)],':black');
    if abs(phi - 2*pi) < 0.0001
        phi = 0;
    end
    text((r*1.2)*(sin(phi)-0.1), (r*1.1)*cos(phi), num2str(rad2deg(phi)));
    if phi == 0
        phi = 2*pi;
    end
end
    
% Draw arrow and put label over it
phi_a = deg2rad(linspace(0, 20, 100));
r_a = r*1.25;
r_a1 = r_a*1.02;
r_a2 = r_a;
r_a3 = r_a*0.98;
phi_a1 =phi_a(end);
phi_a2 =phi_a(end)+deg2rad(3);
phi_a3 =phi_a(end);
r_peak = [r_a1, r_a2, r_a3, r_a1];
phi_peak = [phi_a1, phi_a2, phi_a3, phi_a1];
r_start = [r_a*0.98, r_a*1.02];
phi_start = [phi_a(1) phi_a(1)];
r_text = r_a*1.1;
phi_text = phi_a(end);
r_rad = r*.5;
phi_rad = deg2rad(45); 

plot(r_a*sin(phi_a),r_a*cos(phi_a),'black');
plot(r_peak.*sin(phi_peak),r_peak.*cos(phi_peak),'black');
plot(r_start.*sin(phi_start),r_start.*cos(phi_start),'black');
text(r_text.*sin(phi_text),r_text.*cos(phi_text),myAngle);
text(r_rad.*sin(phi_rad),r_rad.*cos(phi_rad),myRad);

axis(r*[-1 1 -1 1]);

axis off;

end

**Comment by MarvNum** [comment]
I have written it for my masterthesis and no one else is using it, hope you get it working!

**Comment by BambOoxX** [comment]
@egeerardyn I just patched my `matlab2tikz` to handle `polaraxes`. Probably not the best piece of coding ever, but it seems to work quite well. 
I don't know how to start a pull request at the moment, if you have advice, they are welcome.

[matlab2tikz.txt](https://github.com/matlab2tikz/matlab2tikz/files/3570909/matlab2tikz.txt)

**Comment by BambOoxX** [comment]
With this patch a small glitch remains. Polar axes are drawn after standard axes.
